aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1125,6 +1125,28 @@ std::vector<Tensor> unsafe_split(const Tensor& self, int64_t split_size, int64_t
   return result;
 }
 
+std::vector<Tensor> hsplit(const Tensor& self, int64_t split_size) {
+  TORCH_CHECK(self.dim() >= 1, "hsplit requires a tensor with at least 1 dimension")
+  if (self.dim() > 1) {
+    TORCH_CHECK(self.sizes()[1] % split_size == 0, "tensor split does not result in an equal division")
+    return at::tensor_split(self, split_size, 1);
+  }
+  TORCH_CHECK(self.sizes()[0] % split_size == 0, "tensor split does not result in an equal division")
+  return at::tensor_split(self, split_size, 0);
```

Comment:
```suggestion
  int64_t dim = (self.dim() == 1) ? 0 : 1;
  TORCH_CHECK(self.sizes()[dim] % split_size == 0, "tensor split does not result in an equal division")
  return at::tensor_split(self, split_size, dim);
```

Reply:
Also, there might be a better way to tell the user what they can do to avoid the error. Maybe the error message could be something like `"self.size(", dim, ") is not divisible by split_size"`.

Reply:
I like @kurtamohler's suggestion. Might as well populate the values, too with something like "self.size(dim)=<value> is not divisible by split_size=<split_size".